support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -446,10 +447,27 @@ function CheckoutComponent({ geoId }: Props) {
 		useState(true);
 
 	const [billingPostcode, setBillingPostcode] = useState('');
+	const [billingPostcodeError, setBillingPostcodeError] = useState<string>();
 	const [billingLineOne, setBillingLineOne] = useState('');
 	const [billingLineTwo, setBillingLineTwo] = useState('');
 	const [billingCity, setBillingCity] = useState('');
-	const [billingState, setBillingState] = useState('');
+	const [billingStateError, setBillingStateError] = useState<string>();
+	/**
+	 * BillingState selector initialised to undefined to hide
+	 * billingStateError message. formOnSubmit checks and converts to
+	 * empty string to display billingStateError message.
+	 */
+	const [billingState, setBillingState] = useState<string | undefined>(
+		undefined,
+	);
```

Comment:
```suggestion
	const [billingState, setBillingState] = useState<string>();
```

If you don't pass a default value, the type defaults to option AKA `string | undefined` in this case.

Reply:
Yes am trying to start in an `undefined` state for the `billingState` dropdown. If undefined I donot display the error message (you don't want it to pop up when form loads) only after you have selected the `billigState` dropdown and did not selected a state.

Reply:
Super, I think we're in agreement, I was suggesting a simpler syntax for that.

e.g.
```ts
const [optionalA, setOptionalA] = useState<string | undefined>(undefined);
// optionalA is now : string | undefined

// Which can also be expressed as...
const [optionalB, setOptionalB] = useState<string>();
// Because we didn't pass the default argument, optionalB is also of type string | undefined
```

Reply:
Ah I see, will update.